Sources/Bot/Services/MergeService.swift

```diff
@@ -401,6 +405,45 @@ extension MergeService {
 
 // MARK: - System types
 
+extension MergeService {
+    public final class Healthcheck {
+
+        public enum Reason: Error, Equatable {
+            case potentialDeadlock
+        }
+
+        public enum Status: Equatable {
+            case ok
+            case unhealthy(Reason)
+        }
+
+        public let status: Property<Status>
+
+        internal init(
+            state: Signal<State, NoError>,
+            statusChecksTimeout: TimeInterval,
+            scheduler: DateScheduler
+            ) {
+            status = Property(
+                initial: .ok,
+                then: state.combinePrevious()
+                    .skipRepeats { lhs, rhs in
+                        return lhs == rhs
+                    }
+                    .flatMap(.latest) { _, current -> SignalProducer<Status, NoError> in
+                        switch current.status {
+                        case .starting, .idle:
+                            return SignalProducer(value: .ok)
+                        default:
+                            return SignalProducer(value: .unhealthy(.potentialDeadlock))
+                                .delay(1.5 * statusChecksTimeout, on: scheduler)
```

Comment:
can we add comment here explaining this delay?

Reply:
Included @ilyapuchka, let me know if it's clear enough.

Reply:
a lil bit complex but ok =)

Reply:
Yeah it's kind of hard to set-up a logic for this 😕